cirq-google/cirq_google/engine/qcs_notebook.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2019 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import dataclasses
+from typing import Union, Optional
+
+import cirq
+
+from cirq_google import (
+    PhasedFSimEngineSimulator,
+    QuantumEngineSampler,
+    SQRT_ISWAP_INV_PARAMETERS,
+    PhasedFSimCharacterization,
+    Bristlecone,
+    get_engine_sampler,
+    get_engine_device,
+)
+
+
+@dataclasses.dataclass
+class QCSObjectsForNotebook:
+    device: cirq.Device
+    sampler: Union[PhasedFSimEngineSimulator, QuantumEngineSampler]
+    signed_in: bool
+
+    @property
+    def is_sampler(self):
+        return self.sampler is QuantumEngineSampler
```

Comment:
is this the suggested `is_simulator`? `sampler` will always be something that implements the `cirq.Sampler` interface. We should use the term `simulator` and invert the logic

Reply:
I think you also need this to be an `isinstance` check, not `is`. Please also add a short docstring.

Is there a reason why you made this a property rather than another dataclass member?

Reply:
Seems like a derived thing to me, I didn't want to store a redundant variable. I can make it a variable if you want.